internal/http_api/api_request.go

```diff
@@ -37,48 +35,58 @@ func NewDeadlineTransport(timeout time.Duration) *http.Transport {
 			}
 			return &deadlinedConn{timeout, c}, nil
 		},
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
 	}
 	return transport
 }
 
 // NegotiateV1 is a helper function to perform a v1 HTTP request
 // and fallback to parsing the old backwards-compatible response format
-func NegotiateV1(method string, endpoint string, body io.Reader) (*simplejson.Json, error) {
+// storing the result in the value pointed to by v.
+func NegotiateV1(method string, endpoint string, v interface{}) error {
```

Comment:
do we use this for anything other than `GET`, now?  if not, let's drop the `method` param entirely?